nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -130,7 +130,7 @@ let
           ${concatStringsSep " \\\n" ([ "-f qcow2" ]
           ++ optional (cfg.useBootLoader && cfg.useDefaultFilesystems) "-F qcow2 -b ${systemImage}/nixos.qcow2"
           ++ optional (!(cfg.useBootLoader && cfg.useDefaultFilesystems)) "-o size=${toString config.virtualisation.diskSize}M"
-          ++ [ "$NIX_DISK_IMAGE" ])}
+          ++ [ ''"$NIX_DISK_IMAGE"'' ])}
```

Comment:
```suggestion
          ++ [ lib.escapeShellArg "$NIX_DISK_IMAGE" ])}
```

Reply:
Ah, I didn't know that `lib` contains such helper functions.

Anyway, I don't think this works. I check the definition of `escapeShellArg`. The call `escapeShellArg "$NIX_DISK_IMAGE"` would result in the string `'$NIX_DISK_IMAGE'` (notice the **single** quotes) being written into the shell script. The shell variable would then no longer expand due to these single quotes.

I quickly checked this with a local nixpkgs (note: the call of `escapeShellArg "$NIX_DISK_IMAGE"` must be within parenthesis) and indeed, this results in a disk image file with name `$NIX_DISK_IMAGE` (unexpanded) created in the current directory.

But let me know if there is a better way than the two single quotes from my commit to escape the shell variable and maintain expansion, I'm not very familiar with nixpkgs's auxiliary functions.